src/Commands/Self/Plugin/MigrateCommand.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Self\Plugin;
+
+/**
+ * Migrates Terminus 2 plugins.
+ *
+ * @package Pantheon\Terminus\Commands\Self\Plugin
+ */
+class MigrateCommand extends InstallCommand
+{
+    /**
+     * Migrate Terminus 2 plugins.
+     *
+     * @command self:plugin:migrate
+     * @aliases plugin:migrate
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
+     */
+    public function migrate()
+    {
+        $plugins_dir = $this->getConfig()->get('plugins2_dir');
+        if (is_dir($plugins_dir)) {
+            $plugin_dirs = glob($plugins_dir . DIRECTORY_SEPARATOR . '*', GLOB_ONLYDIR);
+            if (!empty($plugin_dirs)) {
+                $projects = [];
+                foreach ($plugin_dirs as $dir) {
+                    $projects[] = $this->getProjectNameFromPath($dir);
+                }
+                if (!empty($projects)) {
```

Comment:
@uberhacker redundant `if (!empty($projects)) {`

Reply:
The code for `$this->getProjectNameFromPath` is not as simple as you might think.  The project name is extracted from composer.json and it's possible a plugin was installed manually without or the file is just missing.  Actually, in the latter case, Terminus will throw an error and stop.

Reply:
If you use `array_map` as in the previous example, then you could remove the empty items via `array_filter`. The empty case does need to be handled one way or another. If you kept the `foreach`, I think you'd wan the `if` inside the loop.

Reply:
Good suggestion!

Reply:
@uberhacker The point was that there is no way for `$projects` to be empty because `$plugin_dirs` is checked to not be empty earlier

Reply:
Just because the directory is there does not necessarily mean a plugin is installed.  As mentioned earlier, `$this->getProjectNameFromPath` looks for a composer.json in the project directory.  If that file is missing or corrupted, the function will be unable to determine the org/repo of the project.

Reply:
@uberhacker Sorry, don't get what you are saying. The `$projects[] = $this->getProjectNameFromPath($dir);` ALWAYS will add at least one item to the `$projects` var because `$plugin_dirs` is not empty making `$projects` not empty

Reply:
This was fixed in the latest version.